corehq/motech/repeater_helpers.py

```diff
@@ -15,6 +15,7 @@ class RepeaterResponse:
     status_code = attr.ib()
     reason = attr.ib()
     text = attr.ib(default="")
+    retry = attr.ib(default=True)
```

Comment:
I'm unsure about this. The benefit, of course, is that this is a simple and obvious way to indicate whether to retry a failed attempt. I really like that.

The bit that feels wrong is that it stops RepeaterResponse from being a Response duck type. It makes me want to find a way to achieve the same outcome, but still have RepeaterResponse look like a Response.

I thought of inventing a new status_code value to denote "Permanent Client Error", like, say, unused code 441. But that wouldn't be as obvious as a "retry" attribute.

... That got me wondering whether I'm thinking about this all wrong. RepeaterResponse was invented so that Repeaters that send multiple API calls per payload can return responses that look like the responses of Repeaters that only send one API call per payload. But maybe I should be thinking from the perspective of the caller, [`Repeater.fire_for_record()`](https://github.com/dimagi/commcare-hq/blob/4ca080cd144087785185f55fe1831bcdb49c64c5/corehq/motech/repeaters/models.py#L515) (Couch) or [`send_request()`](https://github.com/dimagi/commcare-hq/blob/4ca080cd144087785185f55fe1831bcdb49c64c5/corehq/motech/repeaters/models.py#L1301) (future SQL). What does it need to know to make the right choice? We could return a real HttpResponse for some Repeaters, or something that says "retry now", "back off", or "we're done" for Repeaters where that is not applicable, or enough.

But that's a bigger change. Maybe this smaller change is right for now, and we make the bigger change separately.

What do you think about all this?

Reply:
What if all repeaters always returned `RepeaterResponse` (maybe renamed to `RepeaterResult`)? Then it would be a single purpose object that's not pretending to be an HTTP response.

Reply:
:+1: I like that idea a lot.